subsys/bluetooth/mesh/transport.c

```diff
@@ -140,17 +140,32 @@ static int send_unseg(struct bt_mesh_net_tx *tx, struct net_buf_simple *sdu,
 	net_buf_add_mem(buf, sdu->data, sdu->len);
 
 	if (IS_ENABLED(CONFIG_BT_MESH_FRIEND)) {
+		if (!bt_mesh_friend_queue_has_space(tx->sub->net_idx,
+						    tx->src, tx->ctx->addr,
+						    NULL, 1)) {
+			if (BT_MESH_ADDR_IS_UNICAST(tx->ctx->addr)) {
+				BT_ERR("Not enough space in Friend Queue");
+				net_buf_unref(buf);
+				return -ENOBUFS;
+			} else {
+				BT_WARN("No space in Friend Queue");
+				goto send;
+			}
```

Comment:
Hold on, I don't think I get this part. As far as I can tell, this'll make us discard incoming messages for the friend queue once it's full, but we should remove the oldest message in the friend queue to make room for new ones. What am I missing?

Reply:
`bt_mesh_friend_queue_has_space()` only checks if too many buffers are already allocated for other pending segmented messages that we're receiving. That's the only case where we don't go and free up messages to make space for a new one in the queue.

Reply:
@trond-snekvik meaning, if we've committed to receiving a segmented message and are in the process of receiving it then another new message may get discarded.

Reply:
@trond-snekvik see also the code comment at the end of the static `friend_queue_has_space()` function that `bt_mesh_friend_queue_has_space()` calls

Reply:
@trond-snekvik we can argue whether a mechanism to abort pending segmented messages should be implemented, however I don't think either way is prohibited by the spec since neither the old nor the new message has yet reached the friend queue.

Reply:
Riiiight, that's what I was missing 😄 Thanks, that makes sense!

Reply:
I think I did the same choice with SAR sessions in the Nordic Mesh SDK, better to finish the sessions we've already started.